constellations/gemini-admin/src/classes/classEdit.js

```diff
@@ -9,23 +9,44 @@ import { OkayModal } from "../modals/okayModal.js";
 import { YesNoModal } from "../modals/yesnoModal.js";
 
 export class ClassEditPage extends React.Component {
-    state = {
-        isEdit: false,
-
-        // class object
-        oldClassId: "",
-        inputClassKey: "",
-        inputTimeString: "",
-
-        selectProgramId: "",
-        selectSemesterId: "",
-        selectLocationId: "",
-
-        programs: [],
-        semesters: [],
-        locations: [],
-        sessions: [],
-    };
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isEdit: false,
+
+            // class object
+            oldClassId: "",
+            inputClassKey: "",
+            inputTimeString: "",
+
+            selectProgramId: "",
+            selectSemesterId: "",
+            selectLocationId: "",
+            listPrograms: [],
+            listSemesters: [],
+            listLocations: [],
+            listSessionsLocal: [],
+            listSessionsRemote: [],
+
+            // other
+            focusedInput: undefined,
+            fullState: "",
+        };
+
+        this.handleChange = this.handleChange.bind(this);
+
+        this.onClickCancel = this.onClickCancel.bind(this);
+        this.onClickDelete = this.onClickDelete.bind(this);
+        this.onClickSave = this.onClickSave.bind(this);
+
+        this.onModalDeleteConfirm = this.onModalDeleteConfirm.bind(this);
+        this.onModalOkSaved = this.onModalOkSaved.bind(this);
+        this.onModalDismiss = this.onModalDismiss.bind(this);
+
+        this.onAddSessions = this.onAddSessions.bind(this);
+        this.onDeleteSession = this.onDeleteSession.bind(this);
+    }
```

Comment:
Once you keep the original format above, you can get rid of this constructor method